Add docs for delete function and detailed resources section (#26)
wso2-extensions/identity-tools-cli: docs/cli-mode.md

    }   
}
```

> #### Allow Deleting Resources

Reviewer: ```suggestion
#### Allow deleting resources
```

By default, the tool does not delete any resources during export or import. During export, the deletion of a resource in the target environment will not delete the corresponding resource file in the local directory. The file will have to be deleted manually. Similarly, during import, the deletion of a resource file in the local directory will not delete the corresponding resource in the target environment. 
The ```ALLOW_DELETE``` property can be used to override this behaviour and allow the tool to delete resources.

```
{
    "ALLOW_DELETE" : true
}
```

> > **Caution:** Use this property with caution as it can delete unexpected resources if configured incorrectly.

Reviewer: ```suggestion
> **Caution:** Use this property cautiously as it can delete required resources if configured incorrectly.
```

Reviewer: changed `unexpected` --> `required` as unexpected is ambiguous.

    }   
}
```

#### Allow deleting resources
By default, the tool does not delete any resources during export or import. During export, the deletion of a resource in the target environment will not delete the corresponding resource file in the local directory. The file will have to be deleted manually. Similarly, during import, the deletion of a resource file in the local directory will not delete the corresponding resource in the target environment. 

> The ```ALLOW_DELETE``` property can be used to override this behaviour and allow the tool to delete resources.

Reviewer: ```suggestion
The ```ALLOW_DELETE``` property can be used to override this behavior and allow the tool to delete resources.
```

```
{
    "ALLOW_DELETE" : true
}
```

> > **Caution:** Use this property cautiously as it can delete required resources if configured incorrectly.

Reviewer: ```suggestion
> **Caution:** Use this property cautiously, as it can delete required resources if misconfigured.
```

> > If using this config, make sure to exclude the resources that should not be deleted, using the ```EXCLUDE``` property.

Reviewer: ```suggestion
> If using this config, make sure to exclude the resources that should not be deleted using the ```EXCLUDE``` property.
```

```
The ```--config``` flag can be used to provide the path to the env specific config folder that contains the ```serverConfig.json``` and ```toolConfig.json``` files with the details of the environment to which the resources should be imported. If the flag is not provided, the tool looks for the server configurations in the environment variables.

> The ```--inputDir``` flag can be used to provide the path to the local directory where the resource configuration files are stored. If the flag is not provided, the tool looks for the resource configuration files at the current working directory.

Reviewer: ```suggestion
The ```--inputDir``` flag can be used to provide the path to the local directory where the resource configuration files are stored. If the flag is not provided, the tool looks for the resource configuration files in the current working directory.
```

> ## Supported Resource Types

Reviewer: ```suggestion
## Supported resource types
```

The tool supports the following resource types:

### Applications

> The tool supports exporting and importing applications. The exported application configuration files can be found under the ```Applications``` folder in the local directory. If it is required to deploy a new application through the import command of the tool, the new file should be placed under the ```Applications``` folder in the local directory.

Reviewer: ```suggestion
The tool supports exporting and importing applications. The exported application configuration files can be found under the ```Applications``` folder in the local directory. If it is required to deploy a new application through the `import` command of the tool, the new file should be placed under the ```Applications``` folder in the local directory.
```

> Since the system applications: ```Console``` and ```My Account``` are made read-only in WSO2-IS, if it is required to update the console and myaccount applications through the import command, add the following to the ```deployment.toml``` of the WSO2 Identity Server and restart the server.

Reviewer: ```suggestion
Since the ```Console``` and ```My Account``` are read-only system applications in WSO2 Identity Server, if it is required to update these applications through the `import` command, add the following configurations to the ```deployment.toml``` file and restart the server.
```

```
[system_applications]
read_only_apps = []
```
> **Caution:** Be cautious when updating the system applications: ```Console``` and ```My Account``` through the tool, since it will result in unexpected errors in these apps if edited incorrectly. It is recommended to exclude the ```Console```, ```My Account``` and the Management application created for the tool during normal usage, unless it is required to update them through the tool.

> ### Identity Providers

Reviewer: ```suggestion
### Identity providers
```

The tool supports exporting and importing identity providers. The exported identity provider configuration files can be found under the ```IdentityProviders``` folder in the local directory. If it is required to deploy a new identity provider through the import command of the tool, the new file should be placed under the ```IdentityProviders``` folder in the local directory.

The resident identity provider can also be exported into a file named ```LOCAL``` and can be updated by modifying the ```LOCAL``` file and using the import command.
> **Caution:** Be cautious when updating the resident identity provider through the ```LOCAL``` file since it will result in unexpected errors in the server if edited incorrectly. It is recommended to exclude the ```LOCAL``` file during normal usage, unless it is required to update the resident identity provider through the tool.

> ### User Stores

Reviewer: ```suggestion
### User stores
```

The ```--inputDir``` flag can be used to provide the path to the local directory where the resource configuration files are stored. If the flag is not provided, the tool looks for the resource configuration files in the current working directory.

## Supported resource types
The tool supports the following resource types:

### Applications
The tool supports exporting and importing applications. The exported application configuration files can be found under the ```Applications``` folder in the local directory. If it is required to deploy a new application through the `import` command of the tool, the new file should be placed under the ```Applications``` folder in the local directory.

Since the ```Console``` and ```My Account``` are read-only system applications in WSO2 Identity Server, if it is required to update these applications through the `import` command, add the following configurations to the ```deployment.toml``` file and restart the server.

> ```

Reviewer: ```suggestion

```
```

Author: Don't we need this for line 367-368?

[system_applications]
read_only_apps = []
```

> > **Caution:** Be cautious when updating the system applications: ```Console``` and ```My Account``` through the tool, since it will result in unexpected errors in these apps if edited incorrectly. It is recommended to exclude the ```Console```, ```My Account``` and the Management application created for the tool during normal usage, unless it is required to update them through the tool.

Reviewer: ```suggestion

> **Caution:** Be cautious when updating the system applications: ```Console``` and ```My Account``` through the tool, since it will result in unexpected errors in these apps if edited incorrectly. It is recommended to exclude the ```Console```, ```My Account``` and the Management application created for the tool during normal usage, unless it is required to update them through the tool.
```

### Identity providers
The tool supports exporting and importing identity providers. The exported identity provider configuration files can be found under the ```IdentityProviders``` folder in the local directory. If it is required to deploy a new identity provider through the import command of the tool, the new file should be placed under the ```IdentityProviders``` folder in the local directory.

The resident identity provider can also be exported into a file named ```LOCAL``` and can be updated by modifying the ```LOCAL``` file and using the import command.

> > **Caution:** Be cautious when updating the resident identity provider through the ```LOCAL``` file since it will result in unexpected errors in the server if edited incorrectly. It is recommended to exclude the ```LOCAL``` file during normal usage, unless it is required to update the resident identity provider through the tool.

Reviewer: ```suggestion

> **Caution:** Be cautious when updating the resident identity provider through the ```LOCAL``` file since it will result in unexpected errors in the server if edited incorrectly. It is recommended to exclude the ```LOCAL``` file during normal usage unless it is required to update the resident identity provider through the tool.
```
